src/targets/gpu/perfdb.cpp

```diff
@@ -88,6 +89,9 @@ std::string generate_miopen_config(const problem_params& pp)
 
 auto query_miopen_db(const std::string& query)
 {
+    static std::mutex g_db_mutex; // NOLINT
+    const std::lock_guard<std::mutex> lock(g_db_mutex);
+
```

Comment:
why does it require lock for reading into db ?

Reply:
Reads don't have any dependency but it complains the db file is locked when a thread tries to open the  db file while it's being used.

But also, that's a good point, db  should be able to handle multiple thread, I'll try and see if it works.

Reply:
Tried using flags on sqlite call explicitly with various combination but no luck.

Reply:
okay, that's fine.  I was just concerned in case it increased run time since it would do sequential reads.